tests/link.ts

```diff
@@ -0,0 +1,862 @@
+import { assert, expect } from 'chai';
+import * as sinon from 'sinon';
+import { execute, asPromiseWrapper, ApolloLink } from '../src/link';
+import * as Observable from 'zen-observable';
+import MockLink from '../src/mockLink';
+import SetContextLink from '../src/setContextLink';
+import { parse } from 'graphql';
+import {
+  FetchResult,
+  Operation,
+} from '../src/types';
+
+import {
+  testLinkResults,
+} from './utils/testingUtils';
+
+const sampleQuery = `query SampleQuery{
+  stub{
+    id
+  }
+}`;
+
+describe('ApolloLink(abstract class)', () => {
+
+  describe('concat', () => {
+    it('should concat a function', (done) => {
+      const returnOne = new SetContextLink({ add: 1 });
+      const link = returnOne.concat((operation, forward) => Observable.of({ data: operation.context.add }));
```

Comment:
@evanshauser this shows my potential fear with forward :(

I'm assuming the library would not see this as a terminating link, yet forward is not used so it in essence is terminating. I think with copy paste / boilerplate, this could really get people confused